pal/src/map/common.cpp

```diff
@@ -49,7 +49,7 @@ INT W32toUnixAccessControl( IN DWORD flProtect )
         MemAccessControl = PROT_EXEC | PROT_READ | PROT_WRITE;
         break;
     case PAGE_EXECUTE :
-        MemAccessControl = PROT_EXEC;
+        MemAccessControl = PROT_EXEC | PROT_READ; // WinAPI PAGE_EXECUTE also implies readable
```

Comment:
> I am not sure you can distinguish different reads. If you can read, you can read, and it seems like on windows PAGE_EXECUTE grants you read permission?

Isn't this to be an issue when reading from another process ?

> I don't mind us switching to PAGE_EXECUTE_READ to be more explicit as well.

+1 to this. It could be nice to pin down the area we actually need to read and don't affect others?